pynsee/utils/requests_session.py

```diff
@@ -0,0 +1,360 @@
+
+import pynsee
+from pynsee.utils._get_credentials import _get_credentials
+from pynsee.utils._create_insee_folder import _create_insee_folder
+
+logger = logging.getLogger(__name__)
+
+
+class PynseeAPISession(requests.Session):
+    """
+    Session class used to allow specific timeouts and
+    """
+
+    INSEE_API_CODES = {
+        # 200:"Opération réussie",
+        # 301:"Moved Permanently" -> r.headers['location']
+        400: "Bad Request",
+        401: "Unauthorized : token missing",
+        403: "Forbidden : missing subscription to API",  #
+        # 404: "Not Found : no results available",
+        406: "Not acceptable : incorrect 'Accept' header",
+        413: "Too many results, query must be splitted",
+        414: "Request-URI Too Long",
+        # Unused (managed through a specific if/else), kept for memory purpose
+        # 429: "Too Many Requests : allocated quota overloaded",
+        500: "Internal Server Error ",
+        503: "Service Unavailable",
+    }
+
+    def __init__(
+        self,
+        http_proxy: str = None,
+        https_proxy: str = None,
+        sirene_key: str = None,
+    ):
+        """
+        Generates a PynseeAPISession used to perform http(s) requests in pynsee
+
+        If the instance is created without arguments, pynsee will try to get
+        variables from environment; if that fails, it will try to retrieve the
+        latest config (stored on disk).
+
+        In the other case (with at least one not-None argument), credentials
+        stored on disk will be ignored and proxies/sirene key arguments will
+        use said arguments.
+        **This should only be done through pynsee.utils.init_conn**.
+
+        See pynsee.utils.init_conn to create a new configuration.
+
+        Parameters
+        ----------
+        http_proxy : str, optional
+            HTTP proxy. Use this if working behind a corporate proxy and pynsee
+            has not already been configured on the machine. Otherwise, leave
+            this to None. The default is None.
+        https_proxy : str, optional
+            HTTPS proxy. Use this if working behind a corporate proxy and
+            pynsee has not already been configured on the machine. Otherwise,
+            leave this to None. The default is None.
+        sirene_key : str, optional
+            Token generated by the API portal. Use this if you wish to access
+            the SIRENE API and pynsee has not already been configured. The
+            default is None.
+
+        """
+
+        super().__init__()
+        self._mount_adapters()
+
+        if any(x is not None for x in (http_proxy, https_proxy, sirene_key)):
+            # set credentials for a new pynsee connection
+            credentials = {
+                "http_proxy": http_proxy,
+                "https_proxy": https_proxy,
+                "sirene_key": sirene_key,
+            }
+        else:
+            credentials = _get_credentials()
```

Comment:
```suggestion
        credentials = _get_credentials()
        
        if http_proxy is not None:
            credentials["http_proxy"] = http_proxy
        
        if https_proxy is not None:
            credentials["https_proxy"] = https_proxy
        
        if sirene_key is not None:
            credentials["sirene_key"] = sirene_key
```

so that the default values are used (otherwise setting proxy will drop the SIRENE key